homeassistant/components/timer/__init__.py

```diff
@@ -321,6 +329,25 @@ def async_start(self, duration: timedelta | None = None):
         )
         self.async_write_ha_state()
 
+    @callback
+    def async_change(self, duration: timedelta) -> None:
+        """Change duration of a running timer."""
+        if self._listener is None or self._end is None:
+            raise HomeAssistantError(
+                f"Timer {self.entity_id} is not running, only active timers can be changed"
+            )
+
+        self._listener()
+        self._listener = None
+        self._end += duration
+        self._duration += duration
```

Comment:
This is odd and incorrect.

Let's say I have a 120-second running timer, assume 30 seconds are left.

After calling this timer change with +30 seconds, I ended up with 60 seconds left, on a 150 seconds timer? The configuration is not allowed to be changed via service calls, so duration should not be able to change.

This also raises the question if one should be able to go beyond the configured time or not? And what if I add so much negative change value, the timer becomes negative in its time left? Will the point in time become in the past?

Reply:
Will make "corrections" in an update PR and then we can discuss it from that point.
Will send you the link in discord once it's made.
Thanks